### PR TITLE
inmem journal must not reset highestSequenceNr after journal cleanup, #27994

### PR DESCRIPTION
**Pull request by patriknw**
* add tck test (failed for exactly this)
* keep track of highest

### CODE REVIEW

**Comment by patriknw** [comment]
Follow up from https://github.com/akka/akka/pull/27496

File: akka-persistence/src/main/scala/akka/persistence/journal/inmem/InmemJournal.scala [hunk 1]
```diff
@@ -87,16 +87,16 @@ private[persistence] class InmemJournal(cfg: Config) extends AsyncWriteJournal w
 private[persistence] trait InmemMessages {
   // persistenceId -> persistent message
   var messages = Map.empty[String, Vector[PersistentRepr]]
+  // persistenceId -> highest used sequence number
+  private var highestSequenceNumbers = Map.empty[String, Long]
```
Comment (helena) [inline akka-persistence/src/main/scala/akka/persistence/journal/inmem/InmemJournal.scala]: Makes me wonder if we'd need a faster data structure or if this isn't quite on the hot path then no.

Comment (patriknw) [reply akka-persistence/src/main/scala/akka/persistence/journal/inmem/InmemJournal.scala]: inmem journal is only intended for testing so shouldn't be a problem

**Review by chbatey — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

**Comment by patriknw** [comment]
Refs https://github.com/akka/akka/issues/27994

**Pull request merged**